File: Downloads/PetyMate-main/frontend/src/services/productService.js
import api from './api';

export const productService = {
  getProducts: (params) => api.get('/api/products', { params }),
  getProductById: (id) => api.get(`/api/products/${id}`),
  addReview: (id, data) => api.post(`/api/products/${id}/review`, data),
};

File: Downloads/PetyMate-main/backend/src/main/java/com/petymate/service/ProductService.java
package com.petymate.service;

import com.petymate.dto.OrderDto;
import com.petymate.dto.PagedResponse;
import com.petymate.dto.ProductDto;
import org.springframework.data.domain.Pageable;
import java.math.BigDecimal;

public interface ProductService {
    PagedResponse<ProductDto.ProductResponse> getProducts(String category, String speciesTags, String brand,
            BigDecimal minPrice, BigDecimal maxPrice, BigDecimal minRating, Boolean featuredOnly,
            String search, String sortBy, Pageable pageable);
    ProductDto.ProductResponse getProductById(Long id);
    void addReview(Long productId, String email, ProductDto.ReviewRequest request);
}

